client/http-client/src/transport.rs

```diff
@@ -71,9 +55,10 @@ impl HttpTransportClient {
 		}
 
 		let client = match target.scheme_str() {
-			Some("http") => HyperClient::Http(Client::new()),
-			#[cfg(feature = "tls")]
-			Some("https") => {
+			#[cfg(not(feature = "__tls"))]
+			Some("http") => Client::new(),
+			#[cfg(all(feature = "__tls", feature = "native-tls", feature = "webpki-tls"))]
+			Some("https") | Some("http") => {
```

Comment:
for simplicity, just use connector based on the feature flag and yes the https connector is a bit slower on plain http than http connector.